environment.js

```diff
@@ -0,0 +1,26 @@
+var path = require('path');
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var glob = require('glob');
+
+export.ensureDirectory = function(directoryName) {
+  if (!fs.existsSync(directoryName)) {
+    mkdirp.sync(directoryName);
+  }
+};
+
+export.loadSpecifications = function(mocha, directory) {
+   function addSpecs(error, files) {
```

Comment:
Hmmm, well this is a philosophical point as to whether hoisting is a good idea or not. I'd say that that it preferable that the flow of execution and the order of the code as written be the same but maybe if you've internalised hoisting then you might want to focus on the intended execution.

I'm not sure there's a right answer here. Maybe its a style guide question.